apps/challenges/aws_utils.py

```diff
@@ -1294,6 +1294,341 @@ def delete_log_group(log_group_name):
+    client = get_boto3_client("iam", challenge_aws_keys)
+    environment_suffix = "{}-{}".format(challenge_obj.pk, settings.ENVIRONMENT)
+
+    node_group_role_name = "evalai-code-upload-nodegroup-role-{}".format(
+        environment_suffix
+    )
+    # Delete custom ECR all access policy and detach to node_group_role
+    try:
+        # Detach custom ECR policy
+        client.detach_role_policy(
+            RoleName=node_group_role_name,
+            PolicyArn=challenge_evaluation_cluster.ecr_all_access_policy_arn,
+        )
+        client.delete_policy(
+            PolicyArn=challenge_evaluation_cluster.ecr_all_access_policy_arn,
+        )
+    except ClientError as e:
+        logger.exception(e)
+        return
+
+    task_execution_policies = settings.EKS_NODE_GROUP_POLICIES
+    for policy_arn in task_execution_policies:
+        try:
+            # Detach AWS managed EKS worker node policy to the role
+            client.detach_role_policy(
+                RoleName=node_group_role_name,
+                PolicyArn=policy_arn,
+            )
+        except ClientError as e:
+            logger.exception(e)
+            return
+
+    try:
+        client.delete_role(
+            RoleName=node_group_role_name,
+        )
+    except ClientError as e:
+        logger.exception(e)
+        return
+
+    eks_role_name = "evalai-code-upload-eks-role-{}".format(environment_suffix)
+    try:
+        # Detach AWS managed EKS cluster policy to the role
+        client.detach_role_policy(
+            RoleName=eks_role_name,
+            PolicyArn=settings.EKS_CLUSTER_POLICY,
+        )
+    except ClientError as e:
+        logger.exception(e)
+        return
+
+    try:
+        client.delete_role(
+            RoleName=eks_role_name,
+        )
+    except ClientError as e:
+        logger.exception(e)
+        return
+
+    try:
+        serializer = ChallengeEvaluationClusterSerializer(
+            challenge_evaluation_cluster,
+            data={
+                "active": False,
+            },
+            partial=True,
+        )
+        if serializer.is_valid():
+            serializer.save()
+            return serializer.data
+    except Exception as e:
+        logger.exception(e)
+        return
+
+
+def delete_evaluation_clusters(queryset):
+    """
+    The function called by the admin action method to delete all the selected evaluation clusters.
+
+    Calls the delete_setup_eks_cluster method. Before calling, verifies that the evaluation cluster is active.
+
+    Parameters:
+    queryset (<class 'django.db.models.query.QuerySet'>): The queryset of selected challenges in the django admin page.
+
+    Returns:
+    dict: keys-> 'count': the number of evaluation clusters successfully deleted.
+                 'failures': a dict of all the failures with their error messages and the challenge pk
+    """
+    from .models import ChallengeEvaluationCluster
```

Comment:
Why are you importing here and not at the top? Is this going to be a celery task?

Reply:
@KhalidRmb It is imported like this in order to preserve circular imports. The current pipeline of creating cluster also maintains similar code structure.

Reply:
Oh...got it